### PR TITLE
Use xcbuild for xcode-wrapper

### PR DESCRIPTION
**Pull request by paulyoung**
These changes allow node packages to be built when Xcode is not installed.

### CODE REVIEW

**Comment by paulyoung** [comment]
Worth mentioning that I did this because I wasn't sure how to override `xcode-wrapper` any other way.

**Comment by adnelson** [comment]
I might not have a chance to test this out for a bit, but I just wanted to acknowledge it and say thanks! I'll give this a spin asap :)

**Comment by adnelson** [comment]
Sooo unfortunately this fails for me, on the example in the circleci tests which builds a native package. Here are the steps I went through:

1. Build nixfromnpm (from either master or this branch) and make sure the binary is in your path
2. In a fresh directory:
  a. `git clone https://github.com/justadudewhohacks/node-addon-tutorial`
  b. `nixfromnpm -o output -f node-addon-tutorial/VectorExample`
  c. `nix-build output -A nodePackages.nan`

The last step succeeds on `master` but fails on this branch :(

**Comment by paulyoung** [comment]
I'd like to get to the bottom of this because we've been using my fork at the commit used in this PR on our CI since I first opened this. We build for NixOS and macOS on Hydra.

I think this may be relevant to an error I'm seeing after upgrading to macOS Mojave. I started getting the following error locally:

```
dyld: Library not loaded: /usr/lib/system/libsystem_network.dylib
  Referenced from: /nix/store/8ykfqv6jx9jvfhnc4cdygdzg0piy8253-Libsystem-osx-10.11.6/lib/libSystem.B.dylib
  Reason: image not found
```

I've since fixed that for normal nix use but now I see it when trying to open a `nix-shell` even with my fork of `nixfromnpm`.

```
...
building '/nix/store/0aw0k2pzibffd8l71hr29in5wnxmcnwa-remove-references-to.drv'...
dyld: Library not loaded: /usr/lib/system/libsystem_network.dylib
  Referenced from: /nix/store/8ykfqv6jx9jvfhnc4cdygdzg0piy8253-Libsystem-osx-10.11.6/lib/libSystem.B.dylib
  Reason: image not found
builder for '/nix/store/0aw0k2pzibffd8l71hr29in5wnxmcnwa-remove-references-to.drv' failed due to signal 6 (Abort trap: 6)
cannot build derivation '/nix/store/d32g7vmq4lsqj3nnjcg3k7lyv87ycm5y-nixfromnpm-0.13.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/79n3gcx51shhm650rmr7knwkxa9pppky-stdenv-darwin.drv', '/nix/store/d32g7vmq4lsqj3nnjcg3k7lyv87ycm5y-nixfromnpm-0.13.0.drv' failed
```

**Comment by paulyoung** [comment]
Similar issue to https://github.com/reflex-frp/reflex-platform/issues/398

**Comment by paulyoung** [comment]
Actually, please disregard the reflex link. That appears to be something else.

**Comment by paulyoung** [comment]
Apparently I missed the part about pinned nixpkgs. Now doing this instead but having no luck.

```nix
{}:

self: super:

let
  nixfromnpmSrc = self.fetchFromGitHub {
    owner  = "paulyoung"; # Forked from adnelson
    repo   = "nixfromnpm";
    rev    = "ef49d70fffd37aac15ecca0adf1d8839f0840820";
    sha256 = "0knjqqpcm2ssc9p9zaq4bnm6d1403yz4mrnyzyw2lxnby052gyaz";
  };
in
  {
    nixfromnpm = (import "${nixfromnpmSrc}/release.nix" {
      nixpkgs = self.pkgs.src;
    }).nixfromnpm;
  }
```

```
error: Package ‘rcs-5.9.4’ in /nix/store/bhcb0bpfyyq30azmqq5ybmf9rd70jp5n-source/pkgs/applications/version-management/rcs/default.nix:19 is not supported on ‘x86_64-apple-darwin’, refusing to evaluate.
```

**Comment by paulyoung** [comment]
Seems to be the same error as https://github.com/alpmestan/ghc.nix/issues/22#issuecomment-436177043

**Comment by paulyoung** [comment]
I'm going to create a new issue since it seems unrelated to Xcode.

File: release.nix [hunk 1]
```diff
@@ -18,8 +18,6 @@ let
               text-render =
                 haskellPackagesNew.callPackage ./nix/text-render.nix { };
 
-              hnix = haskellPackagesOld.hnix_0_4_0;
-
```
Comment (paulyoung) [inline release.nix]: This is unrelated but problematic for the version of nixpkgs I'm using. I'll address this somehow...

**Comment by paulyoung** [comment]
@adnelson it looks like the build is green now although I didn't really change anything.

**Comment by paulyoung** [comment]
Although it appears to have been green in the beginning...

**Comment by paulyoung** [comment]
Going back to the instructions in https://github.com/adnelson/nixfromnpm/pull/146#issuecomment-437513187, these work for me on macOS Mojave using my fork of `nixfromnpm`

See https://github.com/adnelson/nixfromnpm/compare/master...paulyoung:master for a complete diff.

**Comment by adnelson** [comment]
I really like these changes, because the more code we can remove from nixfromnpm and get into library code, the better :)

But these changes for me are still causing a failure when trying to build `nan` (which we've b`een using as a test balloon for packages with native compilation)

```
⇒  nix-build release.nix -A nixfromnpm
/nix/store/jvwdpn0dqdxwf5jmpjb9lg6hyk7hqdf9-nixfromnpm-0.13.0
⇒  ./result/bin/nixfromnpm -o test-test -p nan
Initializing  /Users/allen/workspace/haskell/nixfromnpm/test-test
Generating node libraries in /Users/allen/workspace/haskell/nixfromnpm/test-test
Resolving nan (>=0.0.0)
Trying to fetch from https://registry.npmjs.org
Querying https://registry.npmjs.org for package nan...
Hitting URI https://registry.npmjs.org/nan

Finished resolving nan 2.12.1
Writing package file at /Users/allen/workspace/haskell/nixfromnpm/test-test/nodePackages/nan/2.12.1.nix
Generating package definition object in /Users/allen/workspace/haskell/nixfromnpm/test-test/nodePackages/default.nix
Package nan@>=0.0.0 built successfully.
⇒  nix-build test-test -A nodePackages.nan
these derivations will be built:
  /nix/store/kmznbn9bfzhkny14g10d1ykgls4a0086-nan-2.12.1-nodejs-8.11.3.drv
building '/nix/store/kmznbn9bfzhkny14g10d1ykgls4a0086-nan-2.12.1-nodejs-8.11.3.drv'...
setVariables
unpacking sources
unpacking source archive /nix/store/mwh1qq5adwbcby68zpc5gzpggxm4m1v8-nan-2.12.1.tgz
source root is package
setting SOURCE_DATE_EPOCH to timestamp 1545166582 of file package/package.json
patching sources
configuring
building
running xcodebuild
error: no project found
builder for '/nix/store/kmznbn9bfzhkny14g10d1ykgls4a0086-nan-2.12.1-nodejs-8.11.3.drv' failed with exit code 1
error: build of '/nix/store/kmznbn9bfzhkny14g10d1ykgls4a0086-nan-2.12.1-nodejs-8.11.3.drv' failed
```

If this is just my system that's a problem, then I don't mind merging this, but...

**Comment by paulyoung** [comment]
@adnelson I'm curious if that works for you using the `master` branch of my fork, and if you're open to the other changes there.

**Comment by adnelson** [comment]
Unfortunately it broke with the same error :( maybe the xcode expression in nixpkgs is system-dependent?

**Comment by paulyoung** [comment]
Thanks for checking. I've confirmed it works on macOS and NixOS so not sure why it's failing for you.

**Comment by adnelson** [comment]
I don't want to stand in the way of this if it's just a problem with my setup, or that of a minority of users. @ixmatus do you have an OSX machine? If so does this work for you?

I'd be willing to merge this and revert it if it causes problems for others.

**Comment by ixmatus** [comment]
I do have an OSX machine, I can try doing the same build you did above to verify.

**Comment by ixmatus** [comment]
@adnelson @paulyoung it built fine for me, I think we should merge this.

**Comment by paulyoung** [comment]
Thanks 🙂

I’ll probably make some PRs for the other changes on my fork.

**Pull request merged**